core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -593,7 +593,7 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     val unauthorizedTopicMetaData = unauthorizedTopics.map(topic => new TopicMetadata(topic, Seq.empty[PartitionMetadata], ErrorMapping.AuthorizationCode))
 
-    val topicMetadata = getTopicMetadata(authorizedTopics, request.securityProtocol)
+    val topicMetadata = if (authorizedTopics.isEmpty) Seq.empty[TopicMetadata] else getTopicMetadata(authorizedTopics, request.securityProtocol)
```

Comment:
Probably out of scope for this PR, but we should really change `getTopicMetadata`'s contract not to assume that empty topics means all topics as it's quite error-prone.